Packages/com.unity.inputsystem/InputSystem/Editor/PropertyDrawers/AssetSearchProviders.cs

```diff
@@ -0,0 +1,89 @@
+
+namespace UnityEngine.InputSystem.Editor
+{
+    internal static class SearchConstants
+    {
+        // SearchFlags : these flags are used to customize how search is performed and how search
+        // results are displayed.
+        internal static readonly SearchFlags SearchFlags = SearchFlags.Sorted |
+            SearchFlags.OpenPicker |
+            SearchFlags.Packages;
+
+        // Search.SearchViewFlags : these flags are used to customize the appearance of the PickerWindow.
+        internal static readonly Search.SearchViewFlags ViewFlags = Search.SearchViewFlags.OpenInBuilderMode |
+            Search.SearchViewFlags.DisableBuilderModeToggle |
+            Search.SearchViewFlags.DisableInspectorPreview |
+            Search.SearchViewFlags.DisableSavedSearchQuery;
+    }
+
+    internal static class AssetSearchProviders
+    {
+        // Note that if this method is annotated with [SearchItemProvider] we would register the provider to
+        // the SearchWindow, we currently skip this since we are mainly interested in using SearchItemProvider as
+        // a picker.
+        private static SearchProvider CreateAssetSearchProvider(string id, string displayName,
+            Func<IEnumerable<Object>> assetProvider, Texture2D thumbnail = null)
+        {
+            return new SearchProvider(id, displayName)
+            {
+                priority = 25,
+                toObject = GetObject,
+                fetchItems = (context, items, provider) => Search(context, provider, assetProvider, thumbnail),
+            };
+        }
+
+        internal static SearchProvider CreateInputActionReferenceSearchProvider()
+        {
+            // Match icon used for sub-assets from importer, if null will use cached icon from asset database instead.
+            // Note that .inputactions has one icon and sub-assets (actions) have another in the importer.
+            return CreateAssetSearchProvider(
+                "InputActionReferenceSearchProvider",
+                "Project-Wide Input Actions",
+                GetInputActionReferenceAssets,
+                (Texture2D)EditorGUIUtility.Load(InputActionImporter.kActionIcon));
+        }
+
+        private static IEnumerable<Object> GetInputActionReferenceAssets()
+        {
+            // Extract actions from actions maps from ProjectWideActionAsset.
+            // Note that we cannot do something like:
+            //  return AssetDatabase.LoadAllAssetsAtPath(ProjectWideActionsAsset.kAssetPath)
+            //      .Where((asset) => (asset is InputActionReference));
+            // Since that would return an outdated asset that is out of sync with ProjectWideActionsAsset.
+            // TODO Understand why this is a problem and update this comment
+            var asset = ProjectWideActionsAsset.GetOrCreate();
+            return (from actionMap in asset.actionMaps from action in actionMap.actions select InputActionReference.Create(action)).ToList();
```

Comment:
Seems we need to merge both the assets and the project wide setting info here.
ATM this is just project wide info

Reply:
Seems in the UI it combines both - I'm not fully clear how that works, but the end result looks good.

Reply:
Not sure I follow? This SearchProvider only handles Project-wide, the "ADB" default provider handles the assets without custom code. The framework / standard solutions handles the combining for us.

Reply:
Yep I missed that in the code, but running the UI I see it. Ignore this.